lib/flight_scheduler/allocation_registry.rb

```diff
@@ -193,6 +204,8 @@ def max_parallel_per_node(job, node, allocations: nil, excluding_jobs: nil, rese
     max_jobs
   end
 
+  # NOTE: This method MUST not be called alone! The AllocationRegistry must be saved in
+  #       tandem with the JobRegistry. Failure to do so can lead to an inconsistent state
```

Comment:
On review I agree with your assessment that there are likely situations where a registry maybe saved individually without leading to an inconsistent state.

However in practice these situations are hard to predict. Your example where a single `node` is removed from an `allocation` is a moot point because the neither registry are saved at this point.

Due to the ad-hoc nature of when the registries are saved, it is not possible to make a guarantee about their states.

As such I have reduced the specification from `MUST NOT` to `SHOULD NOT` and ensured the advice is followed across the code base.

3a5c3c6c4c416179873ff279e04fc74f9a6617e0